README.md

```diff
@@ -516,6 +536,10 @@ python main.py \
 
 The above runs several tasks with `hendrycksTest-*` being MMLU.
 
+## Trainer Controller Framework
+
+For details about the trainer controller framework see [examples/trainer_controller/README.md](examples/trainer_controller/README.md)
```

Comment:
can you move few lines explanation here `A framework for controlling the trainer loop using user-defined rules and metrics.` . This can help with optimal stopping of training loop , where user could specify rules like stop training if loss below a threshold

Reply:
why will anyone open details when they dont even know what is Trainer Controller? :)

Reply:
here please add a example of how to pass the YAML to training loop . 
Example:
Sample YAML can look like this
```
controller-metrics:
  - name: loss
    class: Loss
controllers:
  - name: loss-controller
    triggers:
      - on_log
    rule: "loss < 1.0"
    operations:
      - hfcontrols.should_training_stop
```
To use it with training , command 
```
`accelerate launch \
--num_processes=4
--config_file fixtures/accelerate_fsdp_defaults.yaml \
tuning/sft_trainer.py  \
--model_name_or_path $MODEL_PATH  \
--training_data_path $TRAIN_DATA_PATH  \
--output_dir $OUTPUT_PATH  \
--num_train_epochs 5  \
--per_device_train_batch_size 4  \
--learning_rate 1e-5  \
--response_template "\n### Label:"  \
--dataset_text_field "output" \
--peft_method "None"
---trainer_controller_file = sample_yaml
```

then say for details in structure of YAML and what kind of metrics you can pass , refer to [section ](examples/trainer_controller/README.md)

Reply:
> why will anyone open details when they dont even know what is Trainer Controller? :)

Addressed. We have added a preamble to the trainer controller in the main [readme](https://github.com/foundation-model-stack/fms-hf-tuning/blob/1ca546d5050e1f6d4ba52441343fb04dc2423b89/README.md#trainer-controller-framework).

Reply:
Answered [here](https://github.com/foundation-model-stack/fms-hf-tuning/pull/214#discussion_r1673921949).

Reply:
> here please add a example of how to pass the YAML to training loop . Example: Sample YAML can look like this
> 
> ```
> controller-metrics:
>   - name: loss
>     class: Loss
> controllers:
>   - name: loss-controller
>     triggers:
>       - on_log
>     rule: "loss < 1.0"
>     operations:
>       - hfcontrols.should_training_stop
> ```
> 
> To use it with training , command with parameter needed . Pass parameter `trainer_controller_file ` with YAML file. (even i dont know what is the parameter and how to use from these docs)
> 
> ```
> `accelerate launch \
> --num_processes=4
> --config_file fixtures/accelerate_fsdp_defaults.yaml \
> tuning/sft_trainer.py  \
> --model_name_or_path $MODEL_PATH  \
> --training_data_path $TRAIN_DATA_PATH  \
> --output_dir $OUTPUT_PATH  \
> --num_train_epochs 5  \
> --per_device_train_batch_size 4  \
> --learning_rate 1e-5  \
> --response_template "\n### Label:"  \
> --dataset_text_field "output" \
> --peft_method "None"
> ---trainer_controller_file = sample_yaml
> ```
> 
> then say for details in structure of YAML and what kind of metrics you can pass , refer to [section ](examples/trainer_controller/README.md)

Answered https://github.com/foundation-model-stack/fms-hf-tuning/pull/214#discussion_r1673921949.

Reply:
> can you move few lines explanation here `A framework for controlling the trainer loop using user-defined rules and metrics.` . This can help with optimal stopping of training loop , where user could specify rules like stop training if loss below a threshold

Please check this reply https://github.com/foundation-model-stack/fms-hf-tuning/pull/214#discussion_r1673912149